schemaless/schemaless.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package schemaless allows the creation of schema-less queries.
+// Schema-less queries are that not associated with a YANG schema,
+// it is up to the user to ensure that validity of the path and the parameterized typed.
+// These queries have limited functionality compared to standard queries.
+// Unmarshaling only works if the gNMI server returns the value (or a list entry) in a single gpb.Update,
+// this is the standard behavior for leaves. For non-leaves, this can be resolved by requesting JSON encoding
+// (if supported by the server).
```

Comment:
unmarshalDynamic right now ignores the path of the received datapoints and checks that there is no more than a single datapoint, so does it really work if you query a non-leaf?

Reply:
it's explained in the comment, is the server returns a json blob for a non-leaf (Encoding JSON or JSON_IETF) or Get than it work

Reply:
I see. I wonder whether users will actually use this, but I guess it's possible. If you store at a higher node, and then a lower node, then if you query both, you'll get both back even though they might be overlapping -- I guess this could be documented?

Reply:
it seems like that would depend on the server implementation, if a server unpacks the json to leaves, then it work fine. It if doesn't then two paths would be different